config/base.js

```diff
@@ -65,9 +65,10 @@ const SCHEMA_SUBSCRIBE = Joi.object().keys({
 
 const SCHEMA_CONFIG = Joi.object()
     .keys({
+        template: Joi.string().regex(Regex.FULL_TEMPLATE_NAME_WITH_NAMESPACE),
         version: Joi.number().integer().min(1).max(50),
         annotations: Annotations.annotations,
-        jobs: SCHEMA_JOBS.required(),
+        jobs: SCHEMA_JOBS.when('template', { not: Joi.exist(), then: Joi.required(), otherwise: Joi.optional() }),
```

Comment:
Shouldn't we disallow `jobs` for initial version of `Pipeline Templates` feature?
Also extend the similar schema changes to other fields as per the design?
```suggestion
        jobs: SCHEMA_JOBS.when('template', { not: Joi.exist(), then: Joi.required(), otherwise: Joi.forbidden() }),
```

Reply:
it will just be discarded in config-parser